Docs: forms accessibility cleanup

patrickhlauke opened the pull request:
- expands on what's needed for "faked" controls inside a disabled `<fieldset>`
- centralises the accessibility advice about always providing an accessible name, and mentions some of the possible variants; includes strong suggestion not to rely on placeholder and invisible labels as beyond AT users it can be bad for other user groups
- updated warning about validation, which (due to #28414), is actually worse than the (now improved) browser default validation; expanded the example for server-side to explicitly tie error messages to form fields using `aria-describedby`

this should at least partially be backported (manually, since v4 doesn't have the overview page)

patrickhlauke committed:
Expand on disabled fieldsets and faked buttons

include further advice/information on how to disable faked buttons for keyboard/AT users

patrickhlauke committed:
Centralise accessible name advice in forms overview

seems odd to only mention (separately) label, aria-label etc in input-group and layout. the advice is just as pertinent in other sections like select. checks only skims over this.

moving this, in expanded form, into the overview section itself. adding a specific cross-reference (just because they are easily left with no accname at all) in the checks page.

patrickhlauke committed:
Change warning about accessibility, modify server-side example

- paradoxically, due to our current problems with validation (see #28414) and the fact that browsers seem to have improved in this area for the most part, it's now actually better to use browser-native validation
- added explicit `id` and `aria-describedby` association to at least the server-side form error messages, to show how it should be done properly, and expanded the prose for that explaining this.

XhmikosR commented:
@patrickhlauke this needs a rebase :)

patrickhlauke commented:
rebased @XhmikosR

MartijnCuppens commented on site/content/docs/5.0/forms/validation.md:
```diff
@@ -183,43 +185,43 @@ We recommend using client-side validation, but in case you require server-side v
       <span class="input-group-text" id="inputGroupPrepend3">@</span>
-      <input type="text" class="form-control is-invalid" id="validationServerUsername" aria-describedby="inputGroupPrepend3" required>
-      <div class="invalid-feedback">
+      <input type="text" class="form-control is-invalid" id="validationServerUsername" aria-describedby="inputGroupPrepend3 validationServerUsernameFeedback" required>
```
Not sure `@` is a proper description, shouldn't we move the id of the prefix to the label?

patrickhlauke replied:
is is a proper additional description in that it makes it obvious what sighted users would perceive. keep in mind that this description is in addition to the preceding `<label>` that also still acts on the `<input>`, so it's announced as something like "Username, @, input"

XhmikosR commented:
@patrickhlauke this needs another rebase. Do note that for v4 these will need to be manually backported so please rethink if they need to be backported and if so make a PR against the v4-dev branch or push to my v4-dev-xmr branch.

patrickhlauke commented:
ok so after being told to do a rebase, i tried...and have no idea why now all of a sudden i get all those other extraneous commits in here. sigh.

XhmikosR commented:
Yeah, that's not a rebase :P A rebase should have only your patches.

Anyway, I'll try to fix it tomorrow myself.

patrickhlauke committed:
Replace `.sr-only` with `.visually-hidden` in new addition

patrickhlauke commented:
i think for some reason my local env didn't have the latest main, and was refusing to grab it. had to drop it and refetch it. odd. anyway, i THINK i got it more correctly this time around, let's see...

patrickhlauke commented:
manual v4 partial backport here #31234

XhmikosR reviewed (approved):
LGTM, I'd like a review from @mdo and we squash and merge

patrickhlauke committed:
Merge branch 'main' into v5-docs-forms

patrickhlauke committed:
Merge branch 'main' into v5-docs-forms

mdo committed:
Copy edits for clarity in parenthetical

mdo committed:
Copy and formatting tweaks

- Wordsmithing here and there
- Turns some hyphens into em dashes
- Turns a long running comma separated list into an unordered list
- Rearranges some copy just a bit

mdo commented:
@patrickhlauke Gave this a read through and made some edits. Let me know if anything looks off, then we can merge. Thanks!

patrickhlauke committed:
Merge branch 'main' into v5-docs-forms

patrickhlauke commented:
yup, all looking good @mdo, thanks. merging.
